libriscan/biblios/static/js/homepage.js

```diff
@@ -0,0 +1,298 @@
+  function filterTable(query) {
+    const table = document.querySelector('#all-documents table');
+    const gridContainer = document.getElementById('documents-grid');
+    
+    // Filter table rows
+    if (table) {
+      const rows = table.querySelectorAll('tbody tr');
+      let visibleCount = 0;
+      
+      rows.forEach(row => {
+        if (query === '') {
+          row.style.display = '';
+          visibleCount++;
+          return;
+        }
+        
+        const cells = row.querySelectorAll('td');
+        let rowText = '';
+        
+        // Get text from Document ID, Org, Collection, Series columns
+        cells.forEach((cell, index) => {
+          if (index < 4) { // Only search in first 4 columns
+            rowText += cell.textContent.toLowerCase() + ' ';
+          }
+        });
+        
+        if (rowText.includes(query)) {
+          row.style.display = '';
+          visibleCount++;
+        } else {
+          row.style.display = 'none';
+        }
+      });
+      
+      // Show/hide "no results" message
+      const tableContainer = document.querySelector('#all-documents .overflow-x-auto');
+      let noResultsMsg = document.getElementById('no-filter-results');
+      
+      if (visibleCount === 0 && query !== '') {
+        if (!noResultsMsg) {
+          noResultsMsg = document.createElement('div');
+          noResultsMsg.id = 'no-filter-results';
+          noResultsMsg.className = 'alert alert-info mt-4';
+          noResultsMsg.innerHTML = `
+            <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" class="stroke-current shrink-0 w-6 h-6">
+              <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"></path>
+            </svg>
+            <span>No documents match your search. Try a different term.</span>
+          `;
+          tableContainer.parentElement.appendChild(noResultsMsg);
+        }
+      } else if (noResultsMsg) {
+        noResultsMsg.remove();
+      }
+    }
+    
+    // Filter grid cards
+    if (gridContainer && !gridContainer.classList.contains('hidden')) {
+      const gridCards = gridContainer.querySelectorAll('.card');
+      gridCards.forEach(card => {
+        const text = card.textContent.toLowerCase();
+        if (query === '' || text.includes(query)) {
+          card.style.display = '';
+        } else {
+          card.style.display = 'none';
+        }
+      });
+    }
+  }
+  
+  async function performSearch(query) {
+    try {
+      const data = await LibriscanUtils.fetchJSON(`/api/search/?q=${encodeURIComponent(query)}`);
+      
+      if (data.results && data.results.length > 0) {
+        displaySearchResults(data.results);
+      } else {
+        searchResults.innerHTML = '<div class="p-4 text-center text-base-content/60">No documents found</div>';
+        searchResults.classList.remove('hidden');
```

Comment:
Add showToast for error
```suggestion
      searchResults.classList.remove('hidden');
      if (window.LibriscanUtils?.showToast) {
        window.LibriscanUtils.showToast('Search failed. Please try again.', 'error');
      }
```